### PR TITLE
@types/jsreport-core - implement template registry

### PR DESCRIPTION
**Pull request by newbish**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by ExE-Boss** [comment]
To work around the **TypeScript 4.2** bug:
```json
"lib": ["ESNext"]
```
in `tsconfig.json` to something that doesn’t include `lib.es2020.intl.d.ts`, e.g.:
```json
"lib": ["ES2019"]
```

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jsreport-core ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsreport-core@latest/))
was missing the following properties:
1. Reporter
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 00020a6ce0e72e554159c6c10ed38919d82345f9
</p>

**Comment by newbish** [comment]
@taoqf @pofider would you be able to help with the review?

File: types/jsreport-core/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "esnext"
+            "es2020"
```
Comment (peterblazejewicz) [inline types/jsreport-core/tsconfig.json:5]:
```json
"es2020"
```
for me it looks wrong here, IMO, the `jsreport-core` is using `vars`,
what are these features from es2020 in DT?

Comment (peterblazejewicz) [reply types/jsreport-core/tsconfig.json:5]: Is this lib bug still impacting this definition? (sry, just read)

Comment (newbish) [reply types/jsreport-core/tsconfig.json]: having issues running tests locally so trying here

Comment (newbish) [reply types/jsreport-core/tsconfig.json]: It appears that esnext is working once again ;)

**Comment by peterblazejewicz** [comment]
is that change breaking? Can you take apportunity to update version to 2.0?
https://github.com/jsreport/jsreport-core/blob/master/package.json#L2
Sry for all those questions,
Otherwise looks OK

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@newbish thx!

**Pull request merged**